### DISCUSSION TITLE
[BUG] [SP-API] productType "SOCKS" with variation_theme: SIZE/COLOR the scheme doesn’t have a size field and shows errors on seller central

### DISCUSSION THREAD

**Discussion by rosh3000** [amzn/selling-partner-api-models#3667] [General]
I tired adding a productType "SOCKS" with variation_theme: SIZE/COLOR the scheme doesn’t have a size field only an apperal_size=>size.   

When I added the child variation (via API: ListingsApi putListingsItem method) the product shows missing required field: "size" (apperal_size=>size appears) in my seller central inventory and says: "Based on the variation theme selected for this listing, this attribute is required"  

The parent product variation is showing missing "size" in the "variations" section. At first it only show error and no size column and when I try add another variation (via seller central)

 it shows it as the field size is missing

I tired adding a size attribute to the putListing but obviously it is rejected as there is no size in the scheme

Error for child:
![size_child](https://github.com/amzn/selling-partner-api-docs/assets/7099365/81920cd0-067f-4769-9656-9b234994afd3)

Error for parent:
![size_parent](https://github.com/amzn/selling-partner-api-docs/assets/7099365/37b2ac7a-c561-4c1a-8dbf-9d5612d4603e)

Error for parent when adding another variation on seller central
![size_parent_adding_extra_child](https://github.com/amzn/selling-partner-api-docs/assets/7099365/8c206947-7b7f-4de1-9583-37f3da77778f)

**Comment by wszwc** [comment]
Hello, have you solved this problem?
Your variant theme has color/size selected, but there is no binding relationship with the attributes of outer_size=>size and skirt_size=>size?
Even if you uploaded the upper_size, the error is still reported that size is not included in the attribute?

**Comment by rosh3000** [comment]
Sorry no
